### PR TITLE
Up min SDK to API 19

### LINKED ISSUE #2525: Up min SDK to API 19
Up Minimal SDK to API 19

Min API 19 is covered 95% users and it looks that it doesn't have huge effect on China's users

Steps: 
- [x] check gradle files and up minimal SDK to API 19 
- [x] fix/remove _API_ < 19 checks into Navigation SDK

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Up min SDK to API 19 
Closes #2525 

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -1,10 +1,11 @@
 ext {
 
   androidVersions = [
-      minSdkVersion    : 14,
-      targetSdkVersion : 28,
-      compileSdkVersion: 28,
-      buildToolsVersion: '28.0.3'
+      legacyMinSdkVersion     : 14,
```
Comment (RingerJK) [inline gradle/dependencies.gradle]: keep min SDK 14 for legacy API

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java [hunk 2]
```diff
@@ -446,9 +450,9 @@ private void bind() {
   private void initializeBackground() {
     if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
       int navigationViewBannerBackgroundColor = ThemeSwitcher.retrieveThemeColor(getContext(),
-              R.attr.navigationViewBannerBackground);
+        R.attr.navigationViewBannerBackground);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: For Java code we use [this](https://github.com/mapbox/mapbox-navigation-android/tree/master/config/checkstyle) checkstyle and code style. You can find instructions on how to set both in [Setting up Mapbox checkstyle](https://github.com/mapbox/mapbox-gl-native-android/wiki/Setting-up-Mapbox-checkstyle). The same thing happens 👇

File: libnavigation-util/src/main/java/com/mapbox/navigation/utils/extensions/HandlerThreadEx.kt [hunk 3]
```diff
@@ -2,13 +2,9 @@
 
 package com.mapbox.navigation.utils.extensions
 
-import android.os.Build
 import android.os.HandlerThread
 
+@Deprecated("use HandlerThread.quitSafely() instead", ReplaceWith("this.quitSafely()"))
```
Comment (Guardiola31337) [inline libnavigation-util/src/main/java/com/mapbox/navigation/utils/extensions/HandlerThreadEx.kt]: Do we need this? I believe is not necessary.

File: libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt [hunk 4]
```diff
@@ -26,40 +25,36 @@ class NetworkStatusService(private val applicationContext: Context) {
...
     init {
-        @TargetApi(11)
-        when (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-            true -> {
-                val builder = NetworkRequest.Builder()
-                    .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
-                    .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
-                    .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
-                    .addTransportType(NetworkCapabilities.TRANSPORT_VPN)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (Guardiola31337) [inline libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt]: We should revert these changes, they're not affected by bumping the `minSdkVersion` to `19`.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/AndroidSpeechPlayer.java [hunk 5]
```diff
@@ -131,19 +128,8 @@ private void initializeWithLanguage(Locale language) {
...
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
-      voiceListener.onStart();
-    }
-  }
-
   private void setVoiceListener(final VoiceListener voiceListener) {
     this.voiceListener = voiceListener;
-
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
-      textToSpeech.setOnUtteranceCompletedListener(new Api14UtteranceListener(voiceListener));
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/AndroidSpeechPlayer.java]: We should remove `Api14UtteranceListener` as it's not used anymore.

File: libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt (hunk 4 above)
Comment (RingerJK) [reply libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt]: made it to restrict `@Target` annotation

Comment (Guardiola31337) [reply libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt]: What's the problem with `@Target` annotation?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java (hunk 2 above)
Comment (RingerJK) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: it has been already done, so this style is applied by checkstyle from config *folder*

File: libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt (hunk 4 above)
Comment (RingerJK) [reply libnavigation-util/src/main/java/com/mapbox/navigation/utils/network/NetworkStatusService.kt]: it's just not necessary here, seems clean without this annotation

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java (hunk 2 above)
Comment (RingerJK) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java]: has applied Squad android-java codestyle

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/AndroidSpeechPlayer.java (hunk 5 above)
Comment (RingerJK) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/AndroidSpeechPlayer.java]: done

File: libnavigation-util/src/main/java/com/mapbox/navigation/utils/extensions/HandlerThreadEx.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigation-util/src/main/java/com/mapbox/navigation/utils/extensions/HandlerThreadEx.kt]: done

**Comment by Guardiola31337** [comment]
Also @RingerJK minor comment: clean up / smaller commits like `refactored code style`, `applied squad android-java codestyle`, `revert project's checkstyle` could be rebased for a cleaner history. Could you fixup and rebase when you have a chance 🙏?
